### PR TITLE
[Pools] fix derivation of pool account

### PR DESCRIPTION
**Pull request by Ank4n**
closes https://github.com/paritytech-secops/srlabs_findings/issues/408.
This fixes how ProxyDelegator accounts are derived but may cause issues in Westend since it would use the old derivative accounts. Does not affect Polkadot/Kusama as this pallet is not deployed to them yet.

### CODE REVIEW

**Comment by kianenigma** [comment]
Would be good if you take your learnings from this, and dump them into a concise `frame_account_derivation` reference doc. It should cover:

1. How to generate key-less accounts in FRAME
2. What are the footguns. 
3. When yo use `truncating`.

**Review by kianenigma — APPROVED** [review]
lgtm module whatever is needed to fix wetend.

**Review by gpestana — APPROVED** [review]

File: polkadot/runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -1695,13 +1692,11 @@ pub mod migrations {
 
 	/// Unreleased migrations. Add new ones here:
 	pub type Unreleased = (
-		// Migrate NominationPools to `DelegateStake` adapter. This is unversioned upgrade and
-		// should not be applied yet in Kusama/Polkadot.
-		pallet_nomination_pools::migration::unversioned::DelegationStakeMigration<
```
Comment (Ank4n) [inline polkadot/runtime/westend/src/lib.rs:1700]: The removed migrations are already released.

File: polkadot/runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -1657,11 +1657,8 @@ pub type SignedExtra = (
 );
 
 parameter_types! {
-	// This is the max pools that will be migrated in the runtime upgrade. Westend has more pools
-	// than this, but we want to emulate some non migrated pools. In prod runtimes, if weight is not
-	// a concern, it is recommended to set to (existing pools + 10) to also account for any new
-	// pools getting created before the migration is actually executed.
-	pub const MaxPoolsToMigrate: u32 = 250;
+	/// Bounding number of agent pot accounts to be migrated in a single block.
+	pub const MaxAgentsToMigrate: u32 = 300;
```
Comment (Ank4n) [inline polkadot/runtime/westend/src/lib.rs:1661]: Current count of agents (same as pools) in Westend is 269. `300` should be enough to cover all.

**Pull request merged**